src/TypesService.ts

```diff
@@ -266,6 +291,28 @@ const emitDtsCommon = (
         async ({ config, project }) => {
           const compilerOptions = project.getCompilerOptions()
           const diagnosticsHost = ts.createCompilerHost(compilerOptions, true)
+          const getModuleReference = (
+            importPath: string,
+            sourceFilename: O.Option<string>,
+          ): string =>
+            pipe(
+              O.Do,
+              O.apS('sourceFilename', sourceFilename),
+              O.bind('resolvedFilename', ({ sourceFilename }) =>
+                O.fromNullable(
+                  ts.resolveModuleName(
+                    importPath,
+                    sourceFilename,
+                    compilerOptions,
+                    ts.sys,
+                  ).resolvedModule?.resolvedFileName,
+                ),
+              ),
+              O.map(({ sourceFilename, resolvedFilename }) =>
+                path.relative(path.dirname(sourceFilename), resolvedFilename),
```

Comment:
This took me a bit, but I think I get it.  This will force the file to be explicitly `(relative)/index.js`